### ISSUE TITLE
Release 5.0

### ISSUE THREAD

**Issue by pokoli** [mholt/PapaParse#582]
Hi, 

There are currently three PR that require and API change and should be released on a new 5.0 version:

* #543 
* #500
* #497

On the other hand it also exists #581 which I will prefer to merge by removing support for Node 6.0 as it's not supported by the upstream library. 

So I propose the following: 

* Move the master branch to what will be the new 5.0 version, merging all the PR mentioned above. 
* Create a 4.x branch which will only be used to backport bug fixes.
* Drop support for Node 6.0 on 5.0 version as we can not guarantee it works.
* Close 4.x branch on [Node 6.0 EOL which is April 2019](https://github.com/nodejs/Release)

I will like to have opinion of @dboskovic, and pinging also @mholt in case he want's to add something :)

**Comment by dboskovic** [comment]
@pokoli I don't think that API changes to a library like this should be treated lightly. #543 and #500 both have backwards compatible ways of implementing. #497 seems like something that can be punted into a larger refactor or made a setting for now.

**Comment by pokoli** [comment]
@dboskovic which will be the benefit of implementing them in a backwards compatible way? 

For me transition from 4.x to 5.0 should be straight forward on most of the cases

**Comment by dboskovic** [comment]
Semver only keeps us from breaking people's code *automatically*. At the end of the day, we're making some 100,000 people make a minor change to their code for no actual reason, since we could have done this without breaking the past API. I would want to reserve major versions for things we *absolutely cannot* implement with backwards compatibility or a major overhaul that would provide a significant benefit to drive users to actually upgrade.

As an engineering manager, if I finally slated the engineering hours on my team to have someone review the breaking changes to this dependency - only to find out the change was more or less stylistic for the maintainers and entirely avoidable, while requiring actual work for my team, I'd be pretty annoyed.

**Comment by dboskovic** [comment]
I agree about schedule and all. My point is that it's _unnecessary_ to break the API with these changes, so we're literally creating work for the people who use this library without adding new value.

In my opinion, if the scope of this release is fixes. We should make those fixes as accessible as possible to people rather than being opinionated about the purity of the library. We can add backlog issues to clean up deprecated functions when we have a reason to actually do a major release in the future. I'd love to start planning a V5 and talking through what we could do to add real value and make it exciting for the users of the library.

I don't think the contents of this PR is an ok scope for a major release.

**Comment by pokoli** [comment]
Creating backlog issues will only delay the work for the future. Why we must do it later if we can do it now? 

Which is the problem about having a new major with not big new features? I see one benefit: The upgrade process is easier. 

I'm not against of merging fixes for the supported branch if any PR is provided, that's I Included a maintenance schedule. But fur sure we should only backport bug fixes. New features (as #543) should go to newer versions. 

P.D: Nobody prevents you to merge the features to a custom branch and use them until you are prepared to upgrade to the new version.

**Comment by pokoli** [comment]
First beta version is uploaded to npm, here are the included changes:

**Comment by cvetanov** [comment]
Hi, any updates on releasing V5?

**Comment by pokoli** [comment]
@cvetanov I've just relased it now
